diff -Nru smart-1.4.1.orig/smart/control.py smart-1.4.1/smart/control.py
--- smart-1.4.1.orig/smart/control.py	2011-10-09 10:50:25.000000000 +0200
+++ smart-1.4.1/smart/control.py	2013-07-21 16:43:55.285748421 +0200
@@ -98,9 +98,6 @@
         found = False
         for channel in hooks.call("create-file-channel", filename):
             if channel:
-                if channel.getAlias() in self._channels:
-                    raise Error, _("There's another channel with alias '%s'") \
-                                 % channel.getAlias()
                 self._channels[channel.getAlias()] = channel
                 found = True
         if not found and tarfile.is_tarfile(filename):
