diff -Nru smart-1.3.1/smart/fetcher.py smart-1.3.1.patched//smart/fetcher.py
--- smart-1.3.1/smart/fetcher.py	2010-05-03 13:34:04.000000000 +0200
+++ smart-1.3.1.patched//smart/fetcher.py	2010-07-01 12:03:59.547543543 +0200
@@ -280,7 +280,7 @@
                 else:
                     item.setSucceeded(uncomppath)
             prog.show()
-            time.sleep(0.1)
+            time.sleep(handler._sleep)
         for handler in handlers:
             handler.stop()
         if not progress:
@@ -671,6 +671,7 @@
         self._fetcher = fetcher
         self._queue = []
         self._cancel = False
+        self._sleep = 0.1
 
     def getQueue(self):
         return self._queue
@@ -1495,6 +1496,7 @@
         self._running = False
         self._multi = pycurl.CurlMulti()
         self._lock = thread.allocate_lock()
+        self._sleep = 0
 
     def tick(self):
         import pycurl
