diff -Nru smart-1.4.1.orig/contrib/bash-completion/smart-completion.sh smart-1.4.1/contrib/bash-completion/smart-completion.sh
--- smart-1.4.1.orig/contrib/bash-completion/smart-completion.sh	2010-12-03 23:08:06.000000000 +0100
+++ smart-1.4.1/contrib/bash-completion/smart-completion.sh	2013-03-29 14:23:39.491611374 +0100
@@ -25,7 +25,7 @@
 
     local cur prev opts
 
-    smartdir=$(python -c "import smart; print smart.__file__" \
+    smartdir=$(python2.7 -c "import smart; print smart.__file__" \
 	           | awk '{sub("/__init__.py[c]?","");print}')
 	commands="$(ls ${smartdir}/commands/*.py \
 	            | awk -F '/' '{gsub(/\.py|__init__.py[c]?|\n/,""); print $NF}')"
