diff -Nru smart-1.3.1.orig//smart/backends/rpm/header.py smart-1.3.1/smart/backends/rpm/header.py
--- smart-1.3.1.orig//smart/backends/rpm/header.py	2010-05-03 13:34:04.000000000 +0200
+++ smart-1.3.1/smart/backends/rpm/header.py	2010-08-21 18:31:44.539303425 +0200
@@ -247,8 +247,6 @@
             if h[1106]: # RPMTAG_SOURCEPACKAGE
                 continue
             arch = h[1022] # RPMTAG_ARCH
-            if getArchScore(arch) == 0:
-                continue
 
             name = h[1000] # RPMTAG_NAME
             epoch = h[1003] # RPMTAG_EPOCH
