diff -Nru smart-1.4.1.orig/smart/backends/rpm/header.py smart-1.4.1/smart/backends/rpm/header.py
--- smart-1.4.1.orig/smart/backends/rpm/header.py	2014-12-03 00:25:21.745772999 +0100
+++ smart-1.4.1/smart/backends/rpm/header.py	2014-12-03 00:35:22.090246969 +0100
@@ -348,7 +348,10 @@
                         pass
                     if vi and vi[:2] == "0:":
                         vi = vi[2:]
-                    upgargs.append((Obs, n[i], CM.get(f[i]&CF), vi))
+                    if vi == None:
+                        upgargs.append((Obs, n[i], ">=", "0-0@%s" %(arch)))
+                    else:
+                        upgargs.append((Obs, n[i], CM.get(f[i]&CF), vi))
                 cnfargs.extend(upgargs)
                 upgargs.append(obstup)
             else:
