diff -Nru smart-1.4.1.orig/smart/backends/rpm/metadata.py smart-1.4.1/smart/backends/rpm/metadata.py
--- smart-1.4.1.orig/smart/backends/rpm/metadata.py	2011-10-09 10:50:25.000000000 +0200
+++ smart-1.4.1/smart/backends/rpm/metadata.py	2013-04-22 03:00:27.917326355 +0200
@@ -195,10 +195,7 @@
                         skip = None
 
                 elif tag == ARCH:
-                    if getArchScore(elem.text) == 0:
-                        skip = PACKAGE
-                    else:
-                        arch = elem.text
+                    arch = elem.text
 
                 elif tag == NAME:
                     name = elem.text
diff -Nru smart-1.4.1.orig/smart/backends/rpm/redcarpet.py smart-1.4.1/smart/backends/rpm/redcarpet.py
--- smart-1.4.1.orig/smart/backends/rpm/redcarpet.py	2011-10-09 10:50:25.000000000 +0200
+++ smart-1.4.1/smart/backends/rpm/redcarpet.py	2013-04-22 03:00:41.871155813 +0200
@@ -191,10 +191,7 @@
         self._release = data
 
     def handleArchEnd(self, name, attrs, data):
-        if getArchScore(data) == 0:
-            self._skip = self.PACKAGE
-        else:
-            self._arch = data
+        self._arch = data
 
     def handleSectionEnd(self, name, attrs, data):
         self._info["group"] = data
diff -Nru smart-1.4.1.orig/smart/backends/rpm/synthesis.py smart-1.4.1/smart/backends/rpm/synthesis.py
--- smart-1.4.1.orig/smart/backends/rpm/synthesis.py	2011-10-09 10:50:25.000000000 +0200
+++ smart-1.4.1/smart/backends/rpm/synthesis.py	2013-04-22 03:00:53.395018030 +0200
@@ -265,9 +265,6 @@
                     version = version.replace("-%s%s" % (disttag, distepoch), "")              
                 versionarch = "%s@%s" % (version, arch)
                 
-                if getArchScore(arch) == 0:
-                    continue
-
                 name = "-".join(rpmnameparts[0:releasepos])
 
                 info = {"nvra": element[0],
diff -Nru smart-1.4.1.orig/smart/backends/rpm/yast2.py smart-1.4.1/smart/backends/rpm/yast2.py
--- smart-1.4.1.orig/smart/backends/rpm/yast2.py	2011-10-09 10:50:25.000000000 +0200
+++ smart-1.4.1/smart/backends/rpm/yast2.py	2013-04-22 03:01:02.441911634 +0200
@@ -188,8 +188,6 @@
                     raise Error("Error loading YaST2 channel info. Possibly " \
                                 "corrupted file.\n%s" % self._pkginfofile)
                 
-                if getArchScore(arch) == 0:
-                    return
                 name = nameparts[0]
                 self.curpkgname = name
                 version = nameparts[1]
