--- Wizard.py-old	2011-09-02 07:28:49.000000000 +0100
+++ Wizard.py	2011-09-02 09:47:38.000000000 +0100
@@ -1,13 +1,6 @@
 #!/usr/bin/env python
 # -*- coding: utf-8 -*-
 
-# Munka:	1. Tobbszros csomagkijeloleskor (bP ftp) fagy. Kilepes megy.	# - 
-#		2. Ures ikont ne akarjon olvasni				# - Megoldva
-#		3. Makrok egyszerre lekerdezese					# - Megoldva
-
-#		6. Szálkezeles javitasa						# - 
-#		7. Python-rpm-re atvinni.					# - 
-
 import sys
 import os
 import gtk
@@ -87,6 +80,8 @@
     return newoutput
 
 def IntVersion(Str):
+    if Str[0] in string.letters:
+      return (-1)
     ReturnValue = 0
     i = 0
     l = len(Str)
@@ -211,9 +206,10 @@
         self.Installed = False
         self.new_package = 0
 	try:
-          rpmq = get_rpm_unerrored_output('rpm -q --queryformat \"%{Name}<br>%{Version}<br>%{Release}\" '+self.Name).split('<br>')
-	  self.InstVersion = rpmq[1]
-	  if 'is not installed' not in self.InstVersion:
+          rpmq = get_rpm_unerrored_output('rpm -q --queryformat \"%{Name}<br>%{Version}<br>%{Release}\" '+self.Name)
+	  if 'is not installed' not in rpmq:
+	    rpmq = rpmq.split("<br>")
+      	    self.InstVersion = rpmq[1]
   	    Inst =  IntVersion(self.InstVersion)
   	    Rel = rpmq[2]
   	    Pack =  IntVersion(self.Version)
@@ -482,11 +478,11 @@
                   self.uninstall_button.set_sensitive(True)
                   self.install_button.set_sensitive(False)
                 else:
-                  self.desc.set_markup(_('<b><big>Undefined error</big>\n\n If you\'re versed user, press Details button for details!\n(It\'s where the Install button used to be.)</b>'))
+                  self.desc.set_markup(_('<b><big>Undefined error</big>\n\n If you\'re versed user, press Details button for details!\n(It\'s where the Install button used to be.)</b>'))    
                   self.glade_xml_wizard2_window.get_widget("label159").set_text(_("Force"))
                   self.force_install_mode_flag = True
                   self.uninstall_button.set_sensitive(True)
-                  self.undefined_error_flag = True
+                  self.undefined_error_flag = True
                   self.glade_xml_wizard2_window.get_widget("label43").set_text(_("Details"))
                   self.error_log = 'error 1'
             except:
